src/replicache.ts

```diff
@@ -926,6 +952,22 @@ export class Replicache<MD extends MutatorDefs = {}> {
     return {requestID, syncHead, ok: httpRequestInfo.httpStatusCode === 200};
   }
 
+  protected async _persist(): Promise<void> {
+    await this._ready;
+    const clientID = await this.clientID;
+    return this._persistLock.withLock(() =>
```

Comment:
Should persist handle its own locking?

Reply:
I found this way to be cleaner. `persist` operates on its parameters. If we changed it we would need to pass in the lock and don't know if that would be better.